mbuild/compound.py

```diff
@@ -312,8 +323,205 @@ def particles_by_name(self, name):
+        >>> benzene.name = 'Benzene'
+        >>> filled = mb.fill_box(benzene,
+        ...                      n_compounds=10,
+        ...                      box=[0, 0, 0, 4, 4, 4])
+        >>> filled.label_rigid_bodies(distinct_bodies='Benzene')
+
+        Create a box of semi-rigid benzenes, where each benzene has a unique
+        rigid body ID and only the carbon portion is treated as rigid.
+
+        >>> import mbuild as mb
+        >>> from mbuild.utils.io import get_fn
+        >>> benzene = mb.load(get_fn('benzene.mol2'))
+        >>> benzene.name = 'Benzene'
+        >>> filled = mb.fill_box(benzene,
+        ...                      n_compounds=10,
+        ...                      box=[0, 0, 0, 4, 4, 4])
+        >>> filled.label_rigid_bodies(distinct_bodies='Benzene',
+        ...                           rigid_particles='C')
+
+        """
+        if discrete_bodies is not None:
+            if isinstance(discrete_bodies, string_types):
+                discrete_bodies = [discrete_bodies]
+        if rigid_particles is not None:
+            if isinstance(rigid_particles, string_types):
+                rigid_particles = [rigid_particles]
+
+        if self.root.max_rigid_id is not None:
+            rigid_id = self.root.max_rigid_id + 1
+            warn("{} rigid bodies already exist.  Incrementing 'rigid_id'"
+                 "starting from {}.".format(rigid_id, rigid_id))
+        else:
+            rigid_id = 0
+
+        for successor in self.successors():
+            if discrete_bodies and successor.name not in discrete_bodies:
+                pass
+            else:
+                for particle in successor.particles():
+                    if rigid_particles and particle.name not in rigid_particles:
+                        pass
+                    else:
+                        particle.rigid_id = rigid_id
+                if discrete_bodies:
+                    rigid_id += 1
+
+    def unlabel_rigid_bodies(self):
+        '''Remove all rigid body labels from the Compound '''
+        self._reset_rigid_ids(rigid_id=None)
+
+    def _reset_rigid_ids(self, rigid_id):
+        """Resets the rigid_id of all rigid Particles in a Compound
+
+        Treats the Compound and all successors that already have a rigid_id as
+        a single rigid body.
+        """
+        for particle in self.particles():
+            if particle.rigid_id is not None:
+                if rigid_id is not None:
+                    particle.rigid_id += rigid_id
+                else:
+                    particle.rigid_id = rigid_id
+
+    def _reorder_rigid_ids(self):
+        """Reorder rigid body IDs ensuring consecutiveness.
+
+        Primarily used internally to ensure consecutive rigid_ids following 
+        removal of a Compound.
+
+        """
+        max_rigid = self.max_rigid_id
+        unique_rigid_ids = set([p.rigid_id for p in self.rigid_particles()])
+        unique_rigid_ids.discard(None)
```

Comment:
Good call.  That was left over from the when I had been calling `rigid_ids()` there instead of `rigid_particles()`